cmd/internal/opentelemetry/wrapperMux.go

```diff
@@ -0,0 +1,192 @@
+	traceSamplerParam = flag.Float64("trace_sampler_param", 0, "used for the 'probability' sampler")
+	traceServiceName  = flag.String("trace_name", "cadvisor", "use localEndpoint service name")
+	traceTags         tracingTags
+)
+
+func init() {
+	flag.Var(&traceTags, "trace_tags", "additional tags for tracing (eg. foo=bar)")
+}
+
+func InitTrace(traceEndpoint, collectorCert string) *otlp.Exporter {
+	exporterOptions := []otlp.ExporterOption{otlp.WithAddress(traceEndpoint)}
+
+	if collectorCert != "" {
+		if creds, err := credentials.NewClientTLSFromFile(collectorCert, ""); err != nil {
+			klog.Fatal(err)
+		} else {
+			exporterOptions = append(exporterOptions, otlp.WithTLSCredentials(creds))
+		}
+	} else {
+		exporterOptions = append(exporterOptions, otlp.WithInsecure())
+	}
+
+	if exporter, err := otlp.NewExporter(exporterOptions...); err != nil {
+		klog.Fatal(err)
+	} else {
+		sampler := getTraceSampler(*traceSampler, *traceSamplerParam)
+		providerOpts := []trace.ProviderOption{
+			trace.WithBatcher(exporter),
+			trace.WithConfig(trace.Config{DefaultSampler: sampler}),
+			trace.WithResourceAttributes(getTraceTags(traceTags)...),
+		}
+
+		if traceProvider, err := trace.NewProvider(providerOpts...); err != nil {
+			klog.Fatal(err)
+		} else {
+			global.SetTraceProvider(traceProvider)
+
+			apiB3, traceCtx, corrCtx := apiTrace.B3{SingleHeader: false}, apiTrace.DefaultHTTPPropagator(), correlation.DefaultHTTPPropagator()
+			options := []propagation.Option{propagation.WithInjectors(apiB3, traceCtx, corrCtx),
+				/*                       */ propagation.WithExtractors(apiB3)}
+			global.SetPropagators(propagation.New(options...))
+		}
+		return exporter
+	}
+
+	return nil
+}
+
+func WrapperServerMux(mux *http.ServeMux) *wrapperMux {
+	return &wrapperMux{mux: mux}
+}
+
+type wrapperMux struct {
+	mux *http.ServeMux
+}
+
+func (m *wrapperMux) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	m.mux.ServeHTTP(writer, request)
+}
+
+func (m *wrapperMux) HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) {
+	wH := wrapperHandler{callback: handler}
+	m.mux.Handle(pattern, newOtHttpHandler(&wH, pattern))
+}
+
+func (m *wrapperMux) Handler(r *http.Request) (http.Handler, string) {
+	h, p := m.mux.Handler(r)
+	return newOtHttpHandler(h, p), p
+}
+
+func (m *wrapperMux) Handle(pattern string, handler http.Handler) {
+	m.mux.Handle(pattern, newOtHttpHandler(handler, pattern))
+}
+
+func newOtHttpHandler(h http.Handler, operation string) http.Handler {
+	return othttp.NewHandler(h, operation, othttp.WithFilter(modifyHeader))
+}
+
+type wrapperHandler struct {
+	callback func(http.ResponseWriter, *http.Request)
+}
+
+func (w *wrapperHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	if w.callback != nil {
+		w.callback(writer, request)
+	} else {
+		panic(fmt.Errorf("not yet implemented"))
+	}
+}
+
+func modifyHeader(request *http.Request) bool {
```

Comment:
what is the purpose of this function?

Reply:
If the traceId has only 16 digets, you have to pad it on teh left side with "0".

That is the propose from the opentelemetry-go Gitter.

https://github.com/open-telemetry/opentelemetry-go/issues/686

Reply:
This should be handled in the otel-go b3 propagator.  We don't want every consumer to have to modify trace headers like this.  I hope that we'll be able to come to a quick agreement on how we want to handle 64-bit B3 trace ID values across the OpenTelemetry ecosystem and get a quick fix implemented in the Go propagator.

Reply:
Closing the loop, this was fixed in https://github.com/open-telemetry/opentelemetry-go/pull/698.